students/mayc4t/lesson09/mailroom5.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env
+# #############################################################################
+# Written By: Mayc4t
+# June 3, 2018
+# Self-paced Python
+# lesson 9 -
+# fileID: mailroom5.py
+# #############################################################################
+
+
+import collections
+from collections import *
+
+
+class Donor():
+    # You’ll want a Donor class –
+    # this will hold all the information about the donor
+    # , and have attributes,
+    #            properties,
+    #            and methods to provide access to the donor specific information that is needed.
+    def __init__(self, first, last, donations=None):
+        self._last = last
+        self._first = first
+        self._donations = donations
+        self._num_gift = len(donations)
+        self._avg = sum(donations)/len(donations)
+
+    @property
+    def donations(self):
+        return(self._donations)
+
+    @property
+    def name(self):
+        self._name = "{} {}".format(self._first, self._last)
```

Comment:
You shouldn't be assigning to an instance attribute inside the property method like this. If you `return "{} {}".format(self._first, self._last)` without assigning to `self._name`, then `self.name` will be accessible on the object instance using the same dot syntax as an instance attribute, but the logic used to calculate the value will be nicely encapsulated in the `name` method.